lib/elixir/lib/kernel.ex

```diff
@@ -5182,19 +5182,31 @@ defmodule Kernel do
   defmacro sigil_W(term, modifiers)
 
   defmacro sigil_W({:<<>>, _meta, [string]}, modifiers) when is_binary(string) do
-    split_words(string, modifiers)
+    split_words(string, modifiers, __CALLER__)
   end
 
-  defp split_words(string, []) do
-    split_words(string, [?s])
+  defp split_words(string, [], caller) do
+    split_words(string, [?s], caller)
   end
 
-  defp split_words(string, [mod])
+  defp split_words(string, [mod], caller)
        when mod == ?s or mod == ?a or mod == ?c do
     case is_binary(string) do
       true ->
         parts = String.split(string)
 
+        parts_with_trailing_comma = :lists.filter(&(:binary.last(&1) == ?,), parts)
+
+        if parts_with_trailing_comma != [] do
+          stacktrace = Macro.Env.stacktrace(caller)
+
+          IO.warn(
+            "The sigils ~w/~W do not allow trailing commas at the end of each word. " <>
```

Comment:
```suggestion
            "the sigils ~w/~W do not allow trailing commas at the end of each word. " <>
```

I believe our warning messages start in lowercase. :)

Reply:
The phrasing was yours! I'll lowercase it :)